thruster/src/context/basic_hyper_context.rs

```diff
@@ -294,10 +264,45 @@ impl Context for BasicHyperContext {
     fn remove(&mut self, key: &str) {
         self.headers.remove(key);
     }
+
+    fn status(&mut self, code: u16) {
+        self.status = code;
+    }
 }
 
 impl HasQueryParams for BasicHyperContext {
     fn set_query_params(&mut self, query_params: HashMap<String, String>) {
         self.query_params = query_params;
     }
 }
+
+#[async_trait]
+impl ContextExt for BasicHyperContext {
+    fn get_params(&self) -> &Params {
+        self.hyper_request.as_ref().unwrap().get_params()
+    }
+
+    fn json<T: serde::Serialize>(&mut self, body: &T) -> Result<(), Box<dyn std::error::Error>> {
+        self.set("Content-Type", "application/json");
+        self.body = Body::from(serde_json::to_vec::<T>(body)?);
+
+        Ok(())
+    }
+
+    async fn get_json<T: DeserializeOwned>(&mut self) -> Result<T, Box<dyn std::error::Error>> {
+        let body = self.request_body.take().unwrap_or_else(|| {
+            self.into_owned_request();
+            self.request_body.take().unwrap()
+        });
+
+        serde_json::from_slice::<T>(&hyper::body::to_bytes(body).await?)
+            .map_err(|e| Box::new(e) as Box<dyn std::error::Error>)
+    }
+
+    fn get_req_header<'a>(&'a self, header: &str) -> Option<&'a str> {
```

Comment:
I think "get_" it's implicit, I suggest `req_header` like in `context.req_header("Authorization").unwrap_or("No 'Authorization' present")`.

Reply:
Similar to above, you're absolutely right. The `Context` can't _set_ a req header so this is a good shortening.